plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java

```diff
@@ -374,6 +391,96 @@ protected IStatus validate() {
 			}
 		};
 		dbc.addValidationStatusProvider(validator);
+		BeanProperties.value(ServerSettingsWizardPageModel.PROPERTY_OC_BINARY_STATUS).observe(model)
+				.addValueChangeListener(event -> getShell().pack());
+	}
+
+	private void createRSyncControls(Composite container, ServerSettingsWizardPageModel model, DataBindingContext dbc) {
+		final RSyncValidator rsyncValidator = RSyncValidator.get();
+		Composite composite = new Composite(container, SWT.NONE);
+		GridDataFactory.fillDefaults().applyTo(composite);
+		GridLayoutFactory.fillDefaults().numColumns(2).applyTo(composite);
+
+		ValueBindingBuilder.bind(WidgetProperties.visible().observe(composite))
+				.to(BeanProperties.value(ServerSettingsWizardPageModel.PROPERTY_RSYNC_STATUS).observe(model))
+				.converting(new Converter(RsyncStatus.class, Boolean.class) {
+
+					@Override
+					public Object convert(Object fromObject) {
+						return fromObject != RsyncStatus.OK;
+					}
+
+				}).in(dbc);
+
+		Label label = new Label(composite, SWT.NONE);
+		ValueBindingBuilder.bind(WidgetProperties.image().observe(label))
+				.to(BeanProperties.value(ServerSettingsWizardPageModel.PROPERTY_RSYNC_STATUS).observe(model))
+				.converting(new Converter(RsyncStatus.class, Image.class) {
+
+					@Override
+					public Object convert(Object fromObject) {
+						switch ((RsyncStatus) fromObject) {
+						case OK:
+							return null;
+						default:
+							return JFaceResources.getImage(Dialog.DLG_IMG_MESSAGE_WARNING);
+						}
+					}
+				}).in(dbc);
+		Link link = new Link(composite, SWT.WRAP);
+		ValueBindingBuilder.bind(WidgetProperties.text().observe(link))
+				.to(BeanProperties.value(ServerSettingsWizardPageModel.PROPERTY_RSYNC_STATUS).observe(model))
+				.converting(new Converter(RsyncStatus.class, String.class) {
+
+					@Override
+					public Object convert(Object fromObject) {
+						switch ((RsyncStatus) fromObject) {
+						case OK:
+							return null;
+						default:
+							return ((RsyncStatus) fromObject).getDetailedMessage(rsyncValidator.getBasePath());
+						}
+					}
+				}).in(dbc);
+		GridDataFactory.fillDefaults().hint(600, SWT.DEFAULT).applyTo(link);
+		link.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (REFRESH_LINK_TEXT.equals(e.text)) {
+					new Job("Checking rsync environment") {
```

Comment:
Shouldn't we show the busy cursor while the job is running to inform the user that things are being processed? (can't verify though since the link isnt visible - which is good - on my mac)